craft_platforms/_architectures.py

```diff
@@ -74,3 +77,24 @@ def to_platform_arch(self) -> str:
 _ARCH_TRANSLATIONS_DEB_TO_PLATFORM = {
     deb: platform for platform, deb in _ARCH_TRANSLATIONS_PLATFORM_TO_DEB.items()
 }
+
+
+def get_base_and_architecture(
+    *, architecture: str
+) -> Tuple[Optional[_distro.DistroBase], Union[DebianArchitecture, Literal["all"]]]:
```

Comment:
What about something like...

```suggestion
def parse_base_and_architecture(
    architecture: str, /
) -> Tuple[Optional[_distro.DistroBase], Union[DebianArchitecture, Literal["all"]]]:
```

I don't really like the name `architecture` but don't currently have a better option, but if we make it positional-only we don't have to worry about that name bing part of the API.

Reply:
Agreed on the names being imperfect. It's tough because the API originally worked on `architecture` and `platform` and those same constructs are now `[base:]architecture` and `[base:]platform` and we don't have a good term for those.

Reply:
I took both of your suggestions.